### DISCUSSION TITLE
how to run commands through the user shell with deno?

### DISCUSSION THREAD

**Discussion by joaomelo** [denoland/deno#12511] [Q&A]
### Discussed in https://github.com/denoland/deno/discussions/12493

---

I get you folks closed the original [issue](https://github.com/denoland/deno/issues/12487#event-5483127970) and moved it to discussions. But I think I have a legitimate situation here and the response to the original issue didn't address what was in question or even clarify a mistake from my part. 

I don't want to be rude or bully the project team, far from it. Just really trying to get myself out of a block. 

This time I added a gif in attempt to better illustrate the case.

---

I'm trying to build a  simple CLI tool to run dev scripts with Deno. The goal is to create cross-platform scripts with type safety.

I'm not succeeding with Deno.run(). I think this is due to part of my scripts are based on global packages installed with NPM like, for example, the [firebase CLI tool](https://www.npmjs.com/package/firebase-tools).

If i run something like this:
```js
  const { code } = await Deno.run({
    cmd: ["firebase", "emulators:start"],
  }).status();
```

I get this error
```
error: Uncaught (in promise) NotFound: O sistema não pode encontrar o arquivo especificado. (os error 2)
  const { code } = await Deno.run({
```

Here in live:
![reproduce](https://user-images.githubusercontent.com/5727050/137894758-35e0f9fd-f688-4c93-8846-e655f63c9157.gif)

I found some clarification in this [issue](https://github.com/denoland/deno/issues/5921) about echo. But, I'm afraid if I prefix everything with `"cmd", "/c",` the program will tie itself to the OS. **Is this a valid assumption?**

**Is there another API that enables us to run commands through the current shell** instead of attempt to find binaries in the system directly?

I think in node this is done with something like this:
``` js
    const p = spawn("firebase emulators:start", {
      shell: true,
    });
```

Thanks for the awesome tool. 
JM

**Comment by kitsonk** [comment]
This is still a discussion.  The issue tracker is for issues with the binary, not bug.  The issue wasn't closed, it was moved and it is a "feature" that the original issue gets closed and the discussion opened.  Either continue the conversation in that discussion, or continue it here, in this discussion.
